src/x3f_extract.c

```diff
@@ -309,38 +310,57 @@ int main(int argc, char *argv[])
     }
 
     if (extract_jpg) {
-      if (X3F_OK != x3f_load_data(x3f, x3f_get_thumb_jpeg(x3f))) {
-	x3f_printf(ERR, "Could not load JPEG thumbnail from %s\n", infile);
+      if (X3F_OK != (ret = x3f_load_data(x3f, x3f_get_thumb_jpeg(x3f)))) {
```

Comment:
There's a part of me that really cringes about having if statements that have consequences (all it takes is a later edit to the code to put something in front of this with an 'and' statement and the load won't happen, right?).  But maybe that's my lack of C showing through...